### PR TITLE
[DO NOT SQUASH] Upstream merge September'24

### PR DESCRIPTION
**Pull request by dhernandez0**
- [x] Check this works: https://github.com/ROCm/rocMLIR-internal/issues/1624

closes: https://github.com/ROCm/rocMLIR-internal/issues/1633 https://github.com/ROCm/rocMLIR-internal/issues/1602 https://github.com/ROCm/rocMLIR-internal/issues/1598 https://github.com/ROCm/rocMLIR-internal/issues/1600

### CODE REVIEW

File: external/llvm-project/mlir/test/Target/LLVMIR/rocdl.mlir [hunk 1]
```diff
@@ -368,16 +368,6 @@ llvm.func @rocdl.wmma(%arg0 : vector<8xf32>, %arg1 : vector<16 x f16>, %arg2 : v
   llvm.return %r0 : vector<8xf32>
 }
 
-llvm.func @rocdl.wmma.fp8(%arg0 : vector<2 x i32>, %arg1 : vector<8xf32>) -> vector<8xf32> {
```
Comment (krzysz00) [inline external/llvm-project/mlir/test/Target/LLVMIR/rocdl.mlir]: Why'd this test get removed?

File: mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp [hunk 2]
```diff
@@ -1226,9 +1226,15 @@ LogicalResult AsUnderlyingShapeConverter::matchAndRewrite(
   // in a standard shape. So, applying it to a logically-shaped tensor gets
   // you the tensor in in-memory layout.
   resultType.getStridePermutation(permutation);
+  // TOSA transpose takes i32
+  SmallVector<int32_t, 4> permutationI32;
+  permutationI32.reserve(permutation.size());
+  std::transform(permutation.begin(), permutation.end(), std::back_inserter(permutationI32),
```
Comment (krzysz00) [inline mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp]: This might be `llvm::map_to_vector`

**Review by krzysz00 — COMMENTED** [review]
Minor questions from staring at the diff, otherwise seems fine

File: external/llvm-project/mlir/test/Target/LLVMIR/rocdl.mlir (hunk 1 above)
Comment (dhernandez0) [reply external/llvm-project/mlir/test/Target/LLVMIR/rocdl.mlir]: It was duplicated. I guess this test in upstream PR was in a different line.

File: mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp (hunk 2 above)
Comment (dhernandez0) [reply mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp]: thanks, I've changed it to llvm::map_to_vector

**Review by krzysz00 — CHANGES_REQUESTED** [review]
Please check `ignoreExternalLinting` on your CI runs instead of `clang-format`ing upstream

**Comment by krzysz00** [comment]
Notes:
- `LLVM_PointerTo` in `external/llvm-project//mlir/include/mlir/Dialect/LLVMIR/LLVMOpBase.td ` seems to be stale
- We've landed a few of our PRs (like the code object one) do we want to maybe pull upstream one more time? (But I can understand not doing so)

**Review by krzysz00 — APPROVED** [review]
Since all I've got is trivialities and the update looks clean, approved

**Comment by dhernandez0** [comment]
> Notes:
> 
> * `LLVM_PointerTo` in `external/llvm-project//mlir/include/mlir/Dialect/LLVMIR/LLVMOpBase.td ` seems to be stale
> * We've landed a few of our PRs (like the code object one) do we want to maybe pull upstream one more time? (But I can understand not doing so)

Thanks! I've removed LLVM_PointerTo and LLVM_OpaquePointer.
IMO as this is urgent, I'd rather not do it. We can get those landed PRs in the next upstream merge, right? Unless there's some reason to do it?

**Comment by jerryyin** [comment]
@dhernandez0 Could you run a weekly on this as soon as possible. That's the usual required coverage to run before we do upstream merge.

**Comment by dhernandez0** [comment]
> @dhernandez0 Could you run a weekly on this as soon as possible. That's the usual required coverage to run before we do upstream merge.

@jerryyin Sure. I understand I need to go to Jenkins (PR-1659 -> Build with Parameters), check "weekly" and click on "Build", right? I already did this but I'm just double-checking as it's the first time I run a weekly.

**Comment by krzysz00** [comment]
@dhernandez0 Yeah, we can pick the patches up with the next one, fine by me

**Pull request merged**